hyperon_das_atomdb/adapters/redis_mongo_db.py

```diff
@@ -353,12 +373,26 @@ def get_matched_node_name(self, node_type: str, substring: str) -> str:
         node_type_hash = self._get_atom_type_hash(node_type)
         mongo_filter = {
             FieldNames.COMPOSITE_TYPE_HASH: node_type_hash,
-            FieldNames.NODE_NAME: {'$regex': substring},
+            '$text': {'$search': substring},
         }
         return [
             document[FieldNames.ID_HASH]
             for document in self.mongo_atoms_collection.find(mongo_filter)
         ]
+    
+    def get_node_starting_with(self, node_type: str, startswith: str):
```

Comment:
Thius method should be in the superclass and implemented somehow in each other subclass